p4app-exercises/basic.p4app/main.py

```diff
@@ -50,19 +56,19 @@ def __init__(self, n, **opts):
 
     # Forward to the host connected to this switch
     sw.insertTableEntry(table_name='MyIngress.ipv4_lpm',
-                        match_fields={'hdr.ipv4.dstAddr': ["10.0.0.%d" % i, 32]},
+                        match_fields={'hdr.ipv4.dstAddr': [hostIP(i), 32]},
                         action_name='MyIngress.ipv4_forward',
-                        action_params={'dstAddr': '00:00:00:00:00:%02x' % i,
-                                          'port': 1})
+                        action_params={'dstAddr': hostMAC(i),
+                                       'port': 1})
 
     # Otherwise send the packet to another switch
     for j in range(1, N+1):
         if i == j: continue
         sw.insertTableEntry(table_name='MyIngress.ipv4_lpm',
-                            match_fields={'hdr.ipv4.dstAddr': ["10.0.0.%d" % j, 32]},
+                            match_fields={'hdr.ipv4.dstAddr': [hostIP(j), 32]},
                             action_name='MyIngress.ipv4_forward',
-                            action_params={'dstAddr': '00:00:00:00:ff:%02x' % j,
-                                              'port': getForwardingPort(i, j)})
+                            action_params={'dstAddr': hostMAC(j),
```

Comment:
should be `switchMAC` here (and with `switchMAC(j) != hostMac(j)`)

Reply:
Good catch.  I may have it correct now.